clients/venice-client/src/main/java/com/linkedin/venice/fastclient/DualReadAvroGenericStoreClient.java

```diff
@@ -135,16 +132,17 @@ private static <T> CompletableFuture<T> dualExecute(
    *  Needs to be investigated */
   @Override
   protected CompletableFuture<V> get(GetRequestContext requestContext, K key) throws VeniceClientException {
+    if (requestContext.isTriggeredByBatchGet) {
```

Comment:
Sure. I will add some comments here. The primary reason for this is if a user calls `batchGet` and we end up using `batchGetUsingSingleGet`, the upper layer would have already triggered a dual read on the thin-client and fast-client for the `batchGet` call. So, we don't need to do a dual read on the internal `get` calls as well. This would otherwise make each key be queried twice via thin-client and once using fast-client